pkg/istio-agent/xds_proxy.go

```diff
@@ -184,19 +184,19 @@ func initXdsProxy(ia *Agent) (*XdsProxy, error) {
 	}()
 
 	go proxy.healthChecker.PerformApplicationHealthCheck(func(healthEvent *health.ProbeEvent) {
-		var req *discovery.DiscoveryRequest
+		var req *discovery.DeltaDiscoveryRequest
 		if healthEvent.Healthy {
-			req = &discovery.DiscoveryRequest{TypeUrl: v3.HealthInfoType}
+			req = &discovery.DeltaDiscoveryRequest{TypeUrl: v3.HealthInfoType}
 		} else {
-			req = &discovery.DiscoveryRequest{
+			req = &discovery.DeltaDiscoveryRequest{
```

Comment:
Healthchecks will be sent over delta xds, instead of the current api

Reply:
How does the xds proxy know? Just features.DeltaXds?

Reply:
If its inside of StreamAggregateResources its Sotw. If its in DeltaAggregateResources its delta.

I think here we can just persist *both* request types. Then in the actual stream handlers we chose which one to use (and its impossible to use the wrong one, since the types will not match)